test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -935,6 +935,110 @@ ConstructorTestClass(int i, bool b)
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should include help message in parameter tool tip by multiple match' {
+        Function Test-Function {
+            param (
+                [Parameter(HelpMessage = 'Some help message')]
+                $Foo,
+
+                $Bar
+            )
+        }
+
+        $expected = '[Object] Foo - Some help message'
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches | Where-Object CompletionText -eq '-Foo'
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should retrieve help message from dynamic parameter' {
+        Function Test-Function {
+            [CmdletBinding()]
+            param ()
+            dynamicparam {
+                $attr = [System.Management.Automation.ParameterAttribute]@{
+                    HelpMessage = "Howdy partner"
+                }
+                $attrCollection = [System.Collections.ObjectModel.Collection[System.Attribute]]::new()
+                $attrCollection.Add($attr)
+
+                $dynParam = [System.Management.Automation.RuntimeDefinedParameter]::new('Foo', [string], $attrCollection)
+
+                $paramDictionary = [System.Management.Automation.RuntimeDefinedParameterDictionary]::new()
+                $paramDictionary.Add('Foo', $dynParam)
+                $paramDictionary
+            }
+
+            end {}
+        }
+
+        $expected = '[string] Foo - Howdy partner'
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches | Where-Object CompletionText -eq '-Foo'
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should have type and name for parameter without help message' {
+        Function Test-Function {
+            param (
+                [Parameter()]
+                $WithAttribute,
+
+                $WithoutAttribute
+            )
+        }
+
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches |
+            Where-Object CompletionText -in '-WithAttribute', '-WithoutAttribute' |
+            Sort-Object CompletionText
+        $res.Count | Should -Be 2
+
+        $res.CompletionText[0] | Should -BeExactly '-WithAttribute'
+        $res.ToolTip[0] | Should -BeExactly '[Object] WithAttribute'
+
+        $res.CompletionText[1] | Should -BeExactly '-WithoutAttribute'
+        $res.ToolTip[1] | Should -BeExactly '[Object] WithoutAttribute'
+    }
+
+    It 'Should include help resource message in parameter tool tip by direct match' {
+        $expected = '[string] Activity - Text to describe the activity for which progress is being reported.'
```

Comment:
We can not use a string in tests if it comes from localized resources.
I suggest to check that a string _present_ after `-Activity` in the result but not check it exactly.

Reply:
As in we shouldn't test the value just that it is there or is there some other reason? Is there a reason for this considering the resource is included in PowerShell?

Reply:
If localization is performed in the future, this string may be in a different language and the test will fail. Since there are a huge number of such tests, in order to avoid a lot of work, the team requires that all tests be independent of localization.